app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -83,11 +87,32 @@ class RegisterController @Inject()(val messagesApi: MessagesApi,
                 }, { case (nino, eligibility) ⇒
                   eligibility.result.fold(
                     infos ⇒ {
-                      Logger.error(s"user $nino has missing information: ${infos.missingInfo.mkString(",")}")
+                      val problemDescription = s"user $nino has missing information: ${infos.missingInfo.mkString(",")}"
+                      Logger.error(problemDescription)
+
+                      val eligibilityCheckEventResult = auditConnector.sendEvent(new EligibilityCheckEvent(nino, Some(problemDescription)))
+                      eligibilityCheckEventResult.onFailure {
+                        case e: Throwable => Logger.error(s"Unable to post eligibility check event to audit connector - ${e.getMessage}", e)
+                      }
+
                       Ok(views.html.register.missing_user_info(infos.missingInfo, personalAccountUrl))
                     }, {
-                      case Some(info) ⇒ Ok(views.html.register.confirm_details(info))
-                      case _ ⇒ SeeOther(routes.RegisterController.notEligible().url)
+                      case Some(info) ⇒ {
+                        val eligibilityCheckEventResult = auditConnector.sendEvent(new EligibilityCheckEvent(nino, None))
+                        eligibilityCheckEventResult.onFailure {
+                          case e: Throwable => Logger.error(s"Unable to post eligibility check event to audit connector - ${e.getMessage}", e)
+                        }
+
+                        Ok(views.html.register.confirm_details(info))
+                      }
+                      case _ ⇒ {
+                        val eligibilityCheckEventResult = auditConnector.sendEvent(new EligibilityCheckEvent(nino, Some("Unknown eligibility problem")))
+                        eligibilityCheckEventResult.onFailure {
+                          case e: Throwable => Logger.error(s"Unable to post eligibility check event to audit connector - ${e.getMessage}", e)
+                        }
+
+                        SeeOther(routes.RegisterController.notEligible().url)
+                      }
```

Comment:
i guess , instead of adding audit logic  in each connector , its better to have an Audit Object with sendEvent() function
and just call the method from each controllers.. so error handling goes out of each connector